src/databricks/labs/ucx/cli.py

```diff
@@ -159,17 +159,13 @@ def validate_external_locations(w: WorkspaceClient, prompts: Prompts):
 @ucx.command
 def ensure_assessment_run(w: WorkspaceClient, run_as_collection: bool = False, a: AccountClient | None = None):
     """ensure the assessment job was run on a workspace"""
-    if run_as_collection:
-        if not a:
-            a = AccountClient(product='ucx', product_version=__version__)
-        account_installer = AccountInstaller(a)
-        workspaces_context = account_installer.get_workspace_contexts(w.get_workspace_id())
-        # if running the cmd as a collection, dont wait for each assessment job to finish as that will take long time
-        skip_job_status = True
-    else:
-        skip_job_status = False
-        workspaces_context = [WorkspaceContext(w)]
-    for ctx in workspaces_context:
+    if not a:
+        a = AccountClient(product='ucx', product_version=__version__)
+    account_installer = AccountInstaller(a)
+    workspace_contexts = account_installer.get_workspace_contexts(w, run_as_collection)
+    # if running the cmd as a collection, don't wait for each assessment job to finish as that will take long time
+    skip_job_status = bool(run_as_collection)
```

Comment:
technically, we can launch parallel assessment on all workspaces and wait for all results to finish, but it still may take hours.